src/components/StatsBoxGrid/useStatsBoxGrid.tsx

```diff
@@ -1,259 +1,100 @@
 import { useTranslation } from "next-i18next"
 
-import { GATSBY_FUNCTIONS_PATH } from "../../constants"
-import { Direction } from "../../types"
-import { getData } from "../../utils/cache"
-import { Lang } from "../../utils/languages"
-import {
-  getLocaleForNumberFormat,
-  isLangRightToLeft,
-} from "../../utils/translations"
+import type { AllMetricData, Lang, StatsBoxMetric } from "@/lib/types"
 
-import { RangeSelector } from "./RangeSelector"
-
-export const ranges = ["30d", "90d"] as const
+import { getLocaleForNumberFormat } from "@/lib/utils/translations"
 
-export interface State {
-  value: string
-  data: Array<{ timestamp: number; value: number }>
-  hasError: boolean
-}
+import { RANGES } from "@/lib/constants"
 
-export interface Metric {
-  title: string
-  description: string
-  state: State
-  buttonContainer: JSX.Element
-  range: string
-  apiUrl: string
-  apiProvider: string
-}
+import { RangeSelector } from "./RangeSelector"
 
-interface IFetchEthstoreResponse {
-  data: {
-    day: number
-    effective_balances_sum_wei: number
-  }
+const formatTotalStaked = (amount: number, locale: string): string => {
+  return new Intl.NumberFormat(locale, {
+    notation: "compact",
+    minimumSignificantDigits: 3,
+    maximumSignificantDigits: 4,
+  }).format(amount)
 }
 
-interface IFetchNodeResponse {
-  result: Array<{ UTCDate: number; TotalNodeCount: number }>
+const formatTVL = (tvl: number, locale: string): string => {
+  return new Intl.NumberFormat(locale, {
+    style: "currency",
+    currency: "USD",
+    notation: "compact",
+    minimumSignificantDigits: 3,
+    maximumSignificantDigits: 4,
+  }).format(tvl)
 }
 
-interface IFetchTotalValueLockedResponse {
-  date: string
-  totalLiquidityUSD: number
+const formatTxs = (txs: number, locale: string): string => {
+  return new Intl.NumberFormat(locale, {
+    notation: "compact",
+    minimumSignificantDigits: 3,
+    maximumSignificantDigits: 4,
+  }).format(txs)
 }
 
-interface IFetchTxResponse {
-  unixTimeStamp: string
-  transactionCount: number
+const formatNodes = (nodes: number, locale: string): string => {
+  return new Intl.NumberFormat(locale, {
+    minimumSignificantDigits: 3,
+    maximumSignificantDigits: 4,
+  }).format(nodes)
 }
 
-export const useStatsBoxGrid = () => {
+export const useStatsBoxGrid = ({
+  totalEthStaked,
+  nodeCount,
+  totalValueLocked,
+  txCount,
+}: AllMetricData): StatsBoxMetric[] => {
   const { t } = useTranslation("page-index")
   const { locale } = useRouter()
 
-  const [totalEthStaked, setTotalEthStaked] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
-  const [valueLocked, setValueLocked] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
-  const [txs, setTxs] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
-  const [nodes, setNodes] = useState<State>({
-    data: [],
-    value: "0",
-    hasError: false,
-  })
```

Comment:
No need for `useState` on these four anymore.. we have what we need from `data`, we just format the values and return it with the return object